### PR TITLE
Modify Notification to allow for a new co-teacher invite notification.

### PR DESCRIPTION
**Pull request by lfryemason**
In order to add the new co-teacher workflow, we want to add a co-teacher invite notification. This was designed to have a different icon and slightly different styles to existing notification types. This PR adds the following to the Notification object (see image below) in order to follow the Figma designs linked below:
* An optional tooltip that shows text on hover
* The ability to add styles and color to the optional buttons
* A new 'collaborate' notification type that uses the "users" icon and secondary brand colors

Also adds new colors to the colors list based on design feedback.

image

## Links
[Figma designs](https://www.figma.com/file/fe70qoiGWnV78j9MKfAEdo/Teacher-Experience-UX%2FUI-Improvements?node-id=927%3A3302&mode=dev)

## Testing story
Nothing should change without using the new optional fields. The below definition will show a test notification using all new features. See image above for expected result

```

    <Notification
      type={NotificationType.collaborate}
      iconStyles={styles.icon}
      notice={'Debra invited you to be a co-teacher'}
      details={
        <BodyTwoText style={{marginBottom: 0}}>
          d.zadinsky@cshoolname.org has invited you to co-teach
          <br />
          <StrongText>Intermediate CS - Block 3</StrongText>
        </BodyTwoText>
      }
      dismissible={false}
      buttonsStyles={styles.buttons}
      tooltipText={
        'As a co-teacher, you will be able to manage students in the section, view their work, and track their progress.'
      }
      buttons={[
        {
          text: 'Decline',
          link: 'https://google.com',
          newWindow: true,
          color: Button.ButtonColor.neutralDark,
          style: styles.declineButton,
        },
        {
          text: 'Accept',
          link: 'https://google.com',
          newWindow: true,
          color: Button.ButtonColor.brandSecondaryDefault,
          style: styles.acceptButton,
        },
      ]}
    />
```

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/templates/Notification.jsx [hunk 1]
```diff
@@ -117,17 +124,36 @@ const Notification = ({
...
     <div className="announcement-notification">
       <div style={{...colorStyles, ...mainStyle}}>
         {type !== NotificationType.course && (
-          <div style={{...styles.iconBox, ...colorStyles}}>
+          <div style={{...styles.iconBox, ...colorStyles, ...iconStyles}}>
             <FontAwesome icon={icons[type]} style={styles.icon} />
           </div>
         )}
-        <div style={styles.contentBox}>
+        <div
```
Comment (lfryemason) [inline apps/src/templates/Notification.jsx]: The notification body now accounts for whether it is dismissible and will not have an extra margin at the end if it is not.

File: apps/src/templates/Notification.jsx [hunk 2]
```diff
@@ -145,7 +171,7 @@ const Notification = ({
               )}
             </div>
           </div>
-          <div style={desktop ? null : styles.buttonsMobile}>
+          <div style={desktop ? buttonsStyles : styles.buttonsMobile}>
```
Comment (etaderhold) [inline apps/src/templates/Notification.jsx]: 1) Maybe call this `buttonsStylesDesktop` if it's only applied in the desktop context,
2) Does the need to pass custom styles only apply to desktop or do we need to apply them in a mobile context too?

File: apps/src/templates/Notification.jsx (hunk 1 above)
Comment (etaderhold) [reply apps/src/templates/Notification.jsx]: What's the context for this change? Request from design, existing bug, other? We'll likely have some eyes test diffs if we're changing existing notifications.

Comment (lfryemason) [reply apps/src/templates/Notification.jsx]: Without this, there is a 28px margin on the right side of the notification. This means that the buttons cannot be as close to the right as was originally planned. I'm talking with Molly to see if 28px is acceptable

File: apps/src/templates/Notification.jsx (hunk 2 above)
Comment (lfryemason) [reply apps/src/templates/Notification.jsx]: 1. see 2
2. Will add to both. This was just an oversight

File: apps/src/templates/Notification.jsx (hunk 1 above)
Comment (lfryemason) [reply apps/src/templates/Notification.jsx]: Molly said the 28px is acceptable, so I just removed this change.

**Review by etaderhold — APPROVED** [review]

**Pull request merged**